### DISCUSSION TITLE
Creating a new table

### DISCUSSION THREAD

**Discussion by alper** [SeaQL/sea-orm#481] [Q&A] [answered]
I want to create a new table, so I threw together a Model. Now what do I do?

I can't find any examples of how to create the database table from this nor where to put the code.

**Comment by alper** [comment]
The (very helpful) Axum example has this:

```
pub async fn create_post_table(db: &DbConn) -> Result<ExecResult, DbErr> {
    let stmt = sea_query::Table::create()
        .table(super::post::Entity)
        .if_not_exists()
        .col(
            ColumnDef::new(super::post::Column::Id)
                .integer()
                .not_null()
                .auto_increment()
                .primary_key(),
        )
        .col(
```

But that seem like I'm repeating myself if I already have the model definition.

**Comment by alper** [comment]
The only way around it to move forward quickly that I see is to create the table in Postgres manually and then derive the the model from there using `sea-orm-cli generate` (which as far as I can see does not do much more than that?).

**Comment by alper** [comment, marked as answer]
I could piece this together:

```rust
    let backend = conn.get_database_backend();
    let schema = Schema::new(backend);
    let table_create_statement = schema.create_table_from_entity(FoursquareGmapsMappings);
    let table_create_result = conn.execute(backend.build(&table_create_statement)).await;

    println!("Table create result: {:?}", table_create_result.unwrap());
```

**Reply by billy1624** [reply]
Yep, this is the way of doing it. Before we have migration (#335) in place

**Reply by alper** [reply]
I think this could maybe also be updated in the Axum example?

**Reply by billy1624** [reply]
Good idea!
